src/trytond_factories/action.py

```diff
@@ -0,0 +1,75 @@
+
+__all__ = [
+    'Action',
+    'Keyword',
+    'Report',
+]
+
+import factory
+import factory_trytond
+
+
+class Action(factory_trytond.TrytonFactory):
+    class Meta:
+        model = 'ir.action'
+
+    name = factory.Faker('word')
+    type = factory.Faker('word', ext_word_list=['tree', 'form'])
+    records = 'selected'
+    usage = factory.Faker('word')
+    keywords = factory.RelatedFactoryList(
+        'trytond_factories.action.Keyword',
+        factory_related_name='action',
+        size=1,
+    )
+    icon = factory_trytond.LazySearch(
+        'ir.ui.icon',
+        lambda stub: [('module', '=', 'ir')],
+    )
+
+
+class Keyword(factory_trytond.TrytonFactory):
+    class Meta:
+        model = 'ir.action.keyword'
+
+    keyword = factory.Faker(
+        'word',
+        ext_word_list=[
+            'tree_open',
+            'form_print',
+            'form_action',
+            'form_relate',
+            'graph_open',
+        ]
+    )
+    action = factory.SubFactory('trytond_factories.action.Action')
+
+
+class Report(factory_trytond.TrytonFactory):
+    class Meta:
+        model = 'ir.action.report'
+
+    action = factory.SubFactory('trytond_factories.action.Action')
```

Comment:
AFAIU, Tryton's action system takes good care to create a backing ir.action for every report, window, wizard or url action. So this subfactory could be up to harmful. See the ActionMixin class in trytond/ir/action.py .

A user is never expected to create the ir.action records for a ir.action.whatever they are creating in the UserInterface administration menu and neither should any ir.action.whatever factory!